task/prefetch-dependencies/0.1/prefetch-dependencies.yaml

```diff
@@ -188,6 +188,48 @@ spec:
         echo "$input"
       }
 
+      function inject_rpm_summary_flag
+      {
+        input="$1"
+
+        if [ "$input" == "rpm" ]; then
+          input="$(jq -n '{type: "rpm", include_summary_in_sbom: "true"}')"
+
+        elif is_json "$input" && input_json_has_rpm "$input"; then
+            input="$(jq '
+                      if type == "array" then
+                        map(if .type == "rpm" then
+                        if has("include_summary_in_sbom") | not then
+                          .include_summary_in_sbom = "true"
+                        else
+                        .
+                        end
+                      else
+                        .
+                      end)
+
+                      elif has("packages") then
+                        .packages |= map(if .type == "rpm" then
+                        if has("include_summary_in_sbom") | not then
+                          .include_summary_in_sbom = "true"
+                        else
+                        .
+                        end
+                      else
+                        .
+                      end)
+
+                      else
+                        if has("include_summary_in_sbom") | not then
+                          .include_summary_in_sbom = "true"
+                        else
+                          .
```

Comment:
You shouldn't need this. The 'else .' branch is only strictly needed for map which iterates over all elements in an array and if you don't define `else .` then it'll return `null` for objects that didn't match `map`'s own condition. Here we have a single object.